src/main/scala/io/iohk/ethereum/ledger/BlockPreparator.scala

```diff
@@ -51,61 +51,66 @@ class BlockPreparator(
     *            Pay 20% of it to the treasury contract
     *  2. Miner is payed a reward for the inclusion of ommers
     *  3. Ommers's miners are payed a reward for their inclusion in this block
+    *  4. There is no reward for block with checkpoint
     *
     * @param block the block being processed
     * @param worldStateProxy the initial state
     * @return the state after paying the appropriate reward to who corresponds
     */
+  // scalastyle:off method.length
   private[ledger] def payBlockReward(
       block: Block,
       worldStateProxy: InMemoryWorldStateProxy
   ): InMemoryWorldStateProxy = {
-    val blockNumber = block.header.number
-
-    val minerRewardForBlock = blockRewardCalculator.calculateMiningRewardForBlock(blockNumber)
-    val minerRewardForOmmers =
-      blockRewardCalculator.calculateMiningRewardForOmmers(blockNumber, block.body.uncleNodesList.size)
-
-    val minerAddress = Address(block.header.beneficiary)
-    val treasuryAddress = blockchainConfig.treasuryAddress
-    val existsTreasuryContract = worldStateProxy.getAccount(treasuryAddress).isDefined
-
-    val worldAfterPayingBlockReward =
-      if (block.header.treasuryOptOut.isEmpty || !existsTreasuryContract) {
-        val minerReward = minerRewardForOmmers + minerRewardForBlock
-        val worldAfterMinerReward = increaseAccountBalance(minerAddress, UInt256(minerReward))(worldStateProxy)
-        log.debug(s"Paying block $blockNumber reward of $minerReward to miner with address $minerAddress")
-        worldAfterMinerReward
-      } else if (block.header.treasuryOptOut.get) {
-        val minerReward = minerRewardForOmmers + minerRewardForBlock * MinerRewardPercentageAfterECIP1098 / 100
-        val worldAfterMinerReward = increaseAccountBalance(minerAddress, UInt256(minerReward))(worldStateProxy)
-        log.debug(
-          s"Paying block $blockNumber reward of $minerReward to miner with address $minerAddress, miner opted-out of treasury"
-        )
-        worldAfterMinerReward
-      } else {
-        val minerReward = minerRewardForOmmers + minerRewardForBlock * MinerRewardPercentageAfterECIP1098 / 100
-        val worldAfterMinerReward = increaseAccountBalance(minerAddress, UInt256(minerReward))(worldStateProxy)
-        val treasuryReward = minerRewardForBlock * TreasuryRewardPercentageAfterECIP1098 / 100
-        val worldAfterTreasuryReward =
-          increaseAccountBalance(treasuryAddress, UInt256(treasuryReward))(worldAfterMinerReward)
-
-        log.debug(
-          s"Paying block $blockNumber reward of $minerReward to miner with address $minerAddress" +
-            s"paying treasury reward of $treasuryReward to treasury with address $treasuryAddress"
-        )
-        worldAfterTreasuryReward
-      }
+    if (block.hasCheckpoint) worldStateProxy
```

Comment:
1. I wonder if we should move this condition outside to the `prepareBlock` method, so as to avoid running `persist` on an unchanged world state?

2. Since `BlockExecutionSpec` is possibly not the best place to test that the account is unchanged, can we add an appropriate test to `BlockPreparatorSpec`?